adapters/lora-vertx/src/main/java/org/eclipse/hono/adapter/lora/impl/LoraProtocolAdapter.java

```diff
@@ -202,16 +203,21 @@ void handleProviderRoute(final RoutingContext ctx, final LoraProvider provider)
                     final Buffer payload = uplinkMessage.getPayload();
 
                     Optional.ofNullable(uplinkMessage.getNormalizedData())
-                        .ifPresent(data -> ctx.put(LoraConstants.NORMALIZED_PROPERTIES, data));
+                            .ifPresent(data -> ctx.put(LoraConstants.NORMALIZED_PROPERTIES, data));
 
                     Optional.ofNullable(uplinkMessage.getAdditionalData())
-                        .ifPresent(data -> ctx.put(LoraConstants.ADDITIONAL_DATA, data));
-
-                    final String contentType = String.format(
-                            "%s%s%s",
-                            LoraConstants.CONTENT_TYPE_LORA_BASE,
-                            provider.getProviderName(),
-                            LoraConstants.CONTENT_TYPE_LORA_POST_FIX);
+                            .ifPresent(data -> ctx.put(LoraConstants.ADDITIONAL_DATA, data));
+
+                    final String contentType;
+                    if (payload.length() > 0) {
+                        contentType = String.format(
+                                "%s%s%s",
+                                LoraConstants.CONTENT_TYPE_LORA_BASE,
+                                provider.getProviderName(),
+                                LoraConstants.CONTENT_TYPE_LORA_POST_FIX);
+                    } else {
+                        contentType = EventConstants.CONTENT_TYPE_EMPTY_NOTIFICATION;
```

Comment:
so far we have defined this content type for *events* only. However, the adapter sends all downstream messages using the Telemetry API ...

Reply:
True. However, currently there is way for the LoRaWAN adapter to send events. And, given the fact that LoRaWAN is very conservative what is being sent, I would say that transmissions might be more important, and maybe more events rather then telemetry. In any case, at the moment, a LoRaWAN device doesn't have the ability to send an empty payload. As it is a valid operation on the TTN provider side, I think it should be a valid operation in Hono as well.

Reply:
> However, currently there is way for the LoRaWAN adapter to send events. And, given the fact that LoRaWAN is very conservative what is being sent, I would say that transmissions might be more important, and maybe more events rather then telemetry.

Not sure if I am following. Are you suggesting to let the Lora adapter send events instead of telemetry messages downstream? I am open to the idea. We might also be able to use the ACK flag of uplink messages to determine whether data should be considered an event or telemetry.

> In any case, at the moment, a LoRaWAN device doesn't have the ability to send an empty payload.

You mean because the Lora adapter doesn't support this?

> As it is a valid operation on the TTN provider side, I think it should be a valid operation in Hono as well.

If that is a valid and reasonable use case for Lora devices, I agree.

Reply:
> > However, currently there is way for the LoRaWAN adapter to send events. And, given the fact that LoRaWAN is very conservative what is being sent, I would say that transmissions might be more important, and maybe more events rather then telemetry.
> 
> Not sure if I am following. Are you suggesting to let the Lora adapter send events instead of telemetry messages downstream? I am open to the idea. We might also be able to use the ACK flag of uplink messages to determine whether data should be considered an event or telemetry.

I simply think that at them moment, there is no way of choosing Event over Telemetry. Although Event sound like a reasonable feature in the context of LoRaWAN.

> 
> > In any case, at the moment, a LoRaWAN device doesn't have the ability to send an empty payload.
> 
> You mean because the Lora adapter doesn't support this?

Exactly. I wouldn't simply switch, but as you mentioned above, find a nice way to let the user choose. Either through the URL, or a query parameter of the URL, through the `LoraProvider` interface, the device configuration, a part of the LoraMessage, or a mix of these options.

> 
> > As it is a valid operation on the TTN provider side, I think it should be a valid operation in Hono as well.
> 
> If that is a valid and reasonable use case for Lora devices, I agree.

So I think a reasonable approach would be the finish this PR, and then start a new one, adding "Events" into the picture.

Reply:
I added #2036 to track the Event discussion.